### ISSUE TITLE
Unable to read exposure hdf5 files

### ISSUE THREAD

**Issue by chahank** [CLIMADA-project/climada_python#243]
With the latest changes to the 'crs' handling exposures files created with the old code are not readable anymore. 

```
from climada.entity import Exposures
exp_houses = Exposures()
exp_houses.read_hdf5("EXP_houses_VNM_v01.hdf5")
```

Error message:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-9e97ae9d3693> in <module>
      1 from climada.entity import Exposures
      2 exp_houses = Exposures()
----> 3 exp_houses.read_hdf5("EXP_houses_VNM_v01.hdf5")
      4 
      5 exp_people = Exposures()

~/Documents/Climada/climada_python/climada/entity/exposures/base.py in read_hdf5(self, file_name)
    685                     setattr(self, key, val)
    686                 if key == 'crs':
--> 687                     self.set_crs(val)
    688 
    689     def read_mat(self, file_name, var_names=None):

~/Documents/Climada/climada_python/climada/entity/exposures/base.py in set_crs(self, crs)
    278         if 'geometry' in self.gdf.columns:
    279             try:
--> 280                 self.gdf.set_crs(crs, inplace=True)
    281             except ValueError:
    282                 # restore popped crs and leave

~/opt/anaconda3/envs/climada_210/lib/python3.8/site-packages/geopandas/geodataframe.py in set_crs(self, crs, epsg, inplace, allow_override)
    776         else:
    777             df = self
--> 778         df.geometry = df.geometry.set_crs(
    779             crs=crs, epsg=epsg, allow_override=allow_override, inplace=True
    780         )

~/opt/anaconda3/envs/climada_210/lib/python3.8/site-packages/geopandas/geoseries.py in set_crs(self, crs, epsg, inplace, allow_override)
    483 
    484         if not allow_override and self.crs is not None and not self.crs == crs:
--> 485             raise ValueError(
    486                 "The GeoSeries already has a CRS which is not equal to the passed "
    487                 "CRS. Specify 'allow_override=True' to allow replacing the existing "

ValueError: The GeoSeries already has a CRS which is not equal to the passed CRS. Specify 'allow_override=True' to allow replacing the existing CRS without doing any transformation. If you actually want to transform the geometries, use 'GeoSeries.to_crs' instead.
```

**Comment by tovogt** [comment]
Would you please either share the code that generated the old exposure, or the hdf5 file, or a different minimal example to reproduce the issue?

I noticed that the HDF5-storing of CRS-values is broken anyways. This is what I get with the both the code before and after the changes to the CRS handling:
```
>>> from climada.entity import Exposures
>>> from shapely.geometry import Point
>>> import geopandas as gpd
>>> exp = Exposures(gpd.GeoDataFrame({"value": [0]}, geometry=[Point(0, 0)], crs="epsg:32632"))
...
>>> str(exp.crs)
'epsg:32632'
>>> str(exp.meta['crs'])
'epsg:32632'
>>> str(exp.gdf.crs)
'epsg:32632'
>>> str(exp.gdf.geometry.crs)
'epsg:32632'
>>> exp.write_hdf5("test.hdf5")
...
>>> exp = Exposures()
...
>>> exp.read_hdf5("test.hdf5")
...
>>> str(exp.crs)
'EPSG:4326'
>>> str(exp.meta['crs'])
'epsg:32632'
>>> str(exp.gdf.crs)
'EPSG:4326'
>>> str(exp.gdf.geometry.crs)
'EPSG:4326'
```
Hence, the CRS stored in the `meta` attribute doesn't agree with the ones in the GeoDataFrame.

**Comment by chahank** [comment]
I do not have the code to reproduce the data, but I can share the hdf5 file. I cannot upload it here on github,  I will send it by email.

**Comment by chahank** [comment]
If the CRS-value is not stored correctly in the HDF-5 files, this is rather unfortunate. Do you know already what the cause is?

**Comment by tovogt** [comment]
This issue has been introduced 4 months ago due to the introduction of the `gdf` attribute. Before that, the `crs` attribute was a metadata attribute of GeoDataFrame. Now, it's a `@property`. From the file you have sent me, I guess that this file has been generated before that. And it must have been generated with a geopandas version prior to 0.8, because the `crs` attribute has been renamed to `_crs` and became a `@property` with version 0.8.

The solution for this issue as well as the problem I described above is not that complicated, fortunately: #244

**Comment by emanuel-schmid** [comment]
solved with #244
